src/FINDOLOGIC/Request/Requester.php

```diff
@@ -56,29 +63,30 @@ public static function create($type)
     /**
      * Sends the request with all set params and makes sure that all required params are in place. It respects the
      * alivetest, alivetest timeout and search timeout.
+     *
+     * @throws ServiceNotAliveException
      */
     public function send()
     {
-        $rawParams = $this->getParams();
+        $rawParams = $this->getParam();
         ParameterValidator::requiredParamsAreSet($rawParams);
+        $this->shopkey = $this->getParam('shopkey');
 
-        $alivetestUrl = sprintf(self::FINDOLOGIC_API_URL, $rawParams['shopurl'],
-            self::FINDOLOGIC_ALIVETEST_ACTION);
-        $searchUrl = sprintf(self::FINDOLOGIC_API_URL, $rawParams['shopurl'], $this->action);
+        $alivetestUrl = $this->getAlivetestUrl();
 
-        $client = new Client();
-        try {
-            $res = $client->request($alivetestUrl);
-        } catch (GuzzleException $e) {
-            $res = null;
-            //TODO: Throw own exception but error message as above.
-        }
+        SendRequestHelper::sendAlivetest($alivetestUrl, self::FINDOLOGIC_ALIVETEST_TIMEOUT);
 
-        if ($res !== null && $res->getStatusCode() == 200) {
-            echo $res->getBody();
-        } else {
-            //TODO: Throw exception that the service is not alive.
-        }
+        $requestUrl = $this->getUrlByRequestType();
         //TODO: Send the request. Make sure that the timeout will be respected.
     }
+
+    private function getAlivetestUrl()
+    {
+        return sprintf(self::FINDOLOGIC_API_URL, $this->shopkey, self::FINDOLOGIC_ALIVETEST_ACTION);
```

Comment:
Make the API base URL configurable, e.g. to simplify the local development setup. Ways to do this are:

* Environment variables.
* Parameter for the constructor.
* Config file (maybe not that one).

Reply:
Other things that could benefit from being configurable:

* Search timeout
* Alivetest timeout